task-launcher/src/tasks/shared/trials/afcStimulus.js

```diff
@@ -667,4 +669,23 @@ export const afcCondtional = ({trialType, responseAllowed, promptAboveButtons, t
             }
         }
     }
+}
+
+export const afcStimulusWithTimeoutCondition = ({trialType, responseAllowed, promptAboveButtons, task } = {}) => {
+    return {
+        timeline: [
+            afcStimulus({
+            trialType: trialType,
+            responseAllowed: responseAllowed,
+            promptAboveButtons: promptAboveButtons,
+            task: task
+            })
+        ],
+        conditional_function: () => {
+            // don't play when skipping trials because app is finished
+            if (isMaxTimeoutReached())
```

Comment:
Need brackets and explicit returns. We can also clear the timer here.
```suggestion
            if (isMaxTimeoutReached()) {
              clearTimeout(store.session.get("maxTimerId"));
              return false;
            } else return true;
```

Reply:
Will it be a problem if clearTimeout is called multiple times with the same value?  I think it might be better to clear it once at the end of the app.

Reply:
It might be a problem to call clearTimeout multiple times with the same maxTimerId.  I'd suggest putting it in the TaskFinished trial:

export const taskFinished = {
  type: jsPsychHTMLMultiResponse,
  data: () => {
    return {
      // save_trial: true,
      assessment_stage: 'instructions',
    };
  },
  on_load: () => {
    // timer cleanup
    if (store.session.get("maxTimerId")) { 
      clearTimeout(store.session.get("maxTimerId"));
      clearTimeout(store.session.get("maxTimerId"), null);
    }
  },